server: TestCheckRestartSafe_AllowMinimumQuorum_Pass failed

server.TestCheckRestartSafe_AllowMinimumQuorum_Pass [failed](https://mesolite.cluster.engflow.com/invocations/default/70118ba9-651d-42d3-92c4-b35fce923e3b?testReportRun=1&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cvc2VydmVyOnNlcnZlcl90ZXN0) on master @ [85c941546445790895987c15086c4bae1e14c7c0](https://github.com/cockroachdb/cockroach/commits/85c941546445790895987c15086c4bae1e14c7c0):


```
=== RUN   TestCheckRestartSafe_AllowMinimumQuorum_Pass
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestCheckRestartSafe_AllowMinimumQuorum_Pass604378538
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:143: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    api_v2_test.go:466: 
        	Error Trace:	pkg/server/api_v2_test.go:466
        	Error:      	Should be true
        	Test:       	TestCheckRestartSafe_AllowMinimumQuorum_Pass
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestCheckRestartSafe_AllowMinimumQuorum_Pass604378538
--- FAIL: TestCheckRestartSafe_AllowMinimumQuorum_Pass (41.58s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCheckRestartSafe_AllowMinimumQuorum_Pass.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

server.TestCheckRestartSafe_AllowMinimumQuorum_Pass [failed](https://mesolite.cluster.engflow.com/invocations/default/8671abd7-3d06-4a74-9984-af2e5eec59c2?testReportRun=3&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cvc2VydmVyOnNlcnZlcl90ZXN0) on master @ [7b9fd1a5c621d91b6bcfb6e04482a1a884e3bc24](https://github.com/cockroachdb/cockroach/commits/7b9fd1a5c621d91b6bcfb6e04482a1a884e3bc24):


```
=== RUN   TestCheckRestartSafe_AllowMinimumQuorum_Pass
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestCheckRestartSafe_AllowMinimumQuorum_Pass2259427198
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:143: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    api_v2_test.go:553: 
        	Error Trace:	pkg/server/api_v2_test.go:553
        	            				pkg/server/api_v2_test.go:449
        	Error:      	Received unexpected error:
        	            	rpc error: code = DeadlineExceeded desc = context deadline exceeded
        	Test:       	TestCheckRestartSafe_AllowMinimumQuorum_Pass
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestCheckRestartSafe_AllowMinimumQuorum_Pass2259427198
--- FAIL: TestCheckRestartSafe_AllowMinimumQuorum_Pass (218.06s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCheckRestartSafe_AllowMinimumQuorum_Pass.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

server.TestCheckRestartSafe_AllowMinimumQuorum_Pass [failed](https://mesolite.cluster.engflow.com/invocations/default/2aa50717-44d6-4174-942d-87ef83e9c757?testReportRun=1&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cvc2VydmVyOnNlcnZlcl90ZXN0) on master @ [45f2866eb2aecb262704fc236e6279bc246c8808](https://github.com/cockroachdb/cockroach/commits/45f2866eb2aecb262704fc236e6279bc246c8808):


```
=== RUN   TestCheckRestartSafe_AllowMinimumQuorum_Pass
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestCheckRestartSafe_AllowMinimumQuorum_Pass3165149441
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:143: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    api_v2_test.go:450: 
        	Error Trace:	pkg/server/api_v2_test.go:450
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	Test:       	TestCheckRestartSafe_AllowMinimumQuorum_Pass
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestCheckRestartSafe_AllowMinimumQuorum_Pass3165149441
--- FAIL: TestCheckRestartSafe_AllowMinimumQuorum_Pass (294.15s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCheckRestartSafe_AllowMinimumQuorum_Pass.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

server.TestCheckRestartSafe_AllowMinimumQuorum_Pass [failed](https://mesolite.cluster.engflow.com/invocations/default/f4485e5b-33fb-4cf3-920a-c1a5e9838d9c?testReportRun=2&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cvc2VydmVyOnNlcnZlcl90ZXN0) on master @ [7df163f07fad86b4043f0ac3690591c97f70bf82](https://github.com/cockroachdb/cockroach/commits/7df163f07fad86b4043f0ac3690591c97f70bf82):


```
=== RUN   TestCheckRestartSafe_AllowMinimumQuorum_Pass
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestCheckRestartSafe_AllowMinimumQuorum_Pass4275711394
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:143: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    api_v2_test.go:450: 
        	Error Trace:	pkg/server/api_v2_test.go:450
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	Test:       	TestCheckRestartSafe_AllowMinimumQuorum_Pass
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestCheckRestartSafe_AllowMinimumQuorum_Pass4275711394
--- FAIL: TestCheckRestartSafe_AllowMinimumQuorum_Pass (172.04s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCheckRestartSafe_AllowMinimumQuorum_Pass.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/server/api_v2_test.go
+++ b/pkg/server/api_v2_test.go
@@ -535,7 +535,7 @@ func TestCheckRestartSafe_Integration(t *testing.T) {
 }
 
 func drain(ctx context.Context, ts1 serverutils.TestServerInterface, t *testing.T) error {
-	timeoutCtx, cancel := context.WithTimeout(ctx, 30*time.Second)
+	timeoutCtx, cancel := context.WithTimeout(ctx, testutils.SucceedsSoonDuration())
 	defer cancel()
 
 	err := ts1.DrainClients(ctx)

